File: python/sieve/sieve.py
def primes(limit):
    """
    Generate a list of prime numbers up to a specified limit.

    This function uses the Sieve of Eratosthenes algorithm to find prime numbers
    up to the given `limit`. It returns a list of all prime numbers found.

    Args:
        limit (int): The upper limit for finding prime numbers.

    Returns:
        list: A list containing all prime numbers less than or equal to `limit`.

    Example:
        >>> primes(10)
        [2, 3, 5, 7]  # Prime numbers from 2 to 10
    """
    is_prime = [True] * (limit + 1)

    for i in range(2, int(limit ** 0.5) + 1):
        if is_prime[i]:
            for j in range(i * i, limit + 1, i):
                is_prime[j] = False

    return [i for i in range(2, limit + 1) if is_prime[i]]

File: javascript/sieve/sieve.js
export const primes = (limit) => {
  // Array to hold the prime numbers
  const primesArray = [];

  // Create an array representing the sieve; initialize all values to true initially
  const sieve = Array(limit + 1).fill(true);

  // Loop through numbers starting from 2 up to the square root of the limit
  for (let num = 2; num <= Math.sqrt(limit); num++) {
    // If the number is marked as true (not crossed out)
    if (sieve[num]) {
      // Mark all multiples of the number as false (not prime)
      for (let multiple = num * num; multiple <= limit; multiple += num) {
        sieve[multiple] = false;
      }
    }
  }

  // Loop through the sieve array to collect prime numbers
  for (let i = 2; i <= limit; i++) {
    // If the value at the index is true (marked as prime)
    if (sieve[i]) {
      // Push the index (prime number) into the primesArray
      primesArray.push(i);
    }
  }

  // Return the array containing prime numbers up to the given limit
  return primesArray;
};
